resources/test-runner.mjs

```diff
@@ -77,3 +101,74 @@ export class TestRunner {
         return invoker.start();
     }
 }
+
+export class AsyncTestRunner extends TestRunner {
+    async runTest() {
+        // Prepare all mark labels outside the measuring loop.
+        const suiteName = this.suite.name;
+        const testName = this.test.name;
+        const syncStartLabel = `${suiteName}.${testName}-start`;
+        const syncEndLabel = `${suiteName}.${testName}-sync-end`;
+        const asyncStartLabel = `${suiteName}.${testName}-async-start`;
+        const asyncEndLabel = `${suiteName}.${testName}-async-end`;
+
+        let syncTime;
+        let asyncStartTime;
+        let asyncTime;
+
+        const runSync = async () => {
+            if (this.params.warmupBeforeSync) {
+                performance.mark("warmup-start");
+                const startTime = performance.now();
+                // Infinite loop for the specified ms.
+                while (performance.now() - startTime < this.params.warmupBeforeSync)
+                    continue;
+                performance.mark("warmup-end");
+            }
+            performance.mark(syncStartLabel);
+            const syncStartTime = performance.now();
+            await this.test.run(this.page);
```

Comment:
Can't we just make this part "virtual"?
e.g.
```
class TestRunner {
    async runTest() {
        ...
        performance.mark(syncStartLabel);
        const syncStartTime = performance.now();
        await _runSyncStep(this.test, this.page);
        const syncEndTime = performance.now();
        ...
    }
}
...
class SyncTestRunner extends TestRunner {
    function _runSyncStep(test, page) {
        test.run(page);
    }
}

class AsyncTestRunner extends TestRunner {
    async function _runSyncStep(test, page) {
        await test.run(page);
    }
}
```

Reply:
yeah, that would make this a bit easier to read.